UT compilation in 32-bit systems

I'm encountering errors when compiling the tests for 32-bit systems, @dtrugman did you try it?

Just tried on Ubuntu 16.04 LTS 32bit (puppetlabs/ubuntu-16.04-32-puppet).
CMake configuration:
```
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
```
Saw the following compilation issues:
```
/home/vagrant/pfs/test/modules.cpp:29:33: error: large integer implicitly truncated to unsigned type [-Werror=overflow]
         expected.offset         = 0xffffffffc03eb000;
...
/home/vagrant/pfs/test/netlink_socket.cpp:26:28: error: large integer implicitly truncated to unsigned type [-Werror=overflow]
         expected.skbuff    = 0xffff9f55ba8e8800;
...
/home/vagrant/pfs/test/unix_socket.cpp:24:31: error: large integer implicitly truncated to unsigned type [-Werror=overflow]
         expected.skbuff       = 0xffff8db2f3e09400;
...
/home/vagrant/pfs/test/meminfo.cpp:27:21: error: large integer implicitly truncated to unsigned type [-Werror=overflow]
         amount      = 34359738367;
...
/home/vagrant/pfs/test/net_socket.cpp:43:39: error: large integer implicitly truncated to unsigned type [-Werror=overflow]
         expected.skbuff               = 0xffff9f55b1420800;
...
/home/vagrant/pfs/test/maps.cpp:40:15: error: large integer implicitly truncated to unsigned type [-Werror=overflow]
         start = 0x7f0b476b6000;
```
Compilation was successful when values are trimmed to 32-bit.

And the mem reader test failed as well with the following error:
```
/home/vagrant/pfs/test/mem.cpp:5: FAILED:
due to unexpected exception with message:
  Couldn't read from memory: Invalid argument
```
Not sure what is the underlying reason.

Will handle those.

--- a/include/pfs/mem.hpp
+++ b/include/pfs/mem.hpp
@@ -37,7 +37,7 @@ public:
 
 public: // API
     std::vector<uint8_t> read(const mem_region& region);
-    std::vector<uint8_t> read(off_t offset, size_t len);
+    std::vector<uint8_t> read(loff_t offset, size_t len);
 
 private:
     friend class task;

--- a/src/mem.cpp
+++ b/src/mem.cpp
@@ -47,13 +47,13 @@ std::vector<uint8_t> mem::read(const mem_region& region)
                 region.end_address - region.start_address);
 }
 
-std::vector<uint8_t> mem::read(off_t offset, size_t bytes)
+std::vector<uint8_t> mem::read(loff_t offset, size_t bytes)
 {
     std::vector<uint8_t> buffer(bytes);
 
     struct iovec iov = {&buffer[0], buffer.size()};
 
-    ssize_t bytes_read = preadv(_fd, &iov, 1, offset);
+    ssize_t bytes_read = preadv64(_fd, &iov, 1, offset);
     if (bytes_read == -1)
     {
         throw std::system_error(errno, std::system_category(),

--- a/test/maps.cpp
+++ b/test/maps.cpp
@@ -11,7 +11,11 @@ using namespace pfs::impl::parsers;
 TEST_CASE("Parse corrupted maps", "[task][maps]")
 {
     // Missing last token (inode)
+#if defined(ARCH_64BIT)
     std::string line = "559037183000-559037184000 rw-p 00185000 fd:00";
+#elif defined(ARCH_32BIT)
+    std::string line = "08153000-08158000 rw-p 0010a000 fc:00";
+#endif
 
     REQUIRE_THROWS_AS(parse_maps_line(line), pfs::parser_error);
 }
@@ -37,8 +41,13 @@ TEST_CASE("Parse maps", "[task][maps]")
 
     SECTION("Anonymous")
     {
+#if defined(ARCH_64BIT)
         start = 0x7f0b476b6000;
         end   = 0x7f0b476b7000;
+#elif defined(ARCH_32BIT)
+        start = 0x08158000;
+        end   = 0x0815d000;
+#endif
 
         can_read    = true;
         can_write   = false;
@@ -57,8 +66,13 @@ TEST_CASE("Parse maps", "[task][maps]")
 
     SECTION("Special")
     {
+#if defined(ARCH_64BIT)
         start = 0xffffffffff600000;
         end   = 0xffffffffff601000;
+#elif defined(ARCH_32BIT)
+        start = 0xbfbd7000;
+        end   = 0xbfbf8000;
+#endif
 
         can_read    = true;
         can_write   = false;
@@ -77,8 +91,13 @@ TEST_CASE("Parse maps", "[task][maps]")
 
     SECTION("Shared object")
     {
+#if defined(ARCH_64BIT)
         start = 0x7f0b476c6000;
         end   = 0x7f0b476c7000;
+#elif defined(ARCH_32BIT)
+        start = 0xb75b7000;
+        end   = 0xb7766000;
+#endif
 
         can_read    = true;
         can_write   = false;
@@ -97,8 +116,13 @@ TEST_CASE("Parse maps", "[task][maps]")
 
     SECTION("Deleted")
     {
+#if defined(ARCH_64BIT)
         start = 0x7fcbe4bc0000;
         end   = 0x7f0b476c6000;
+#elif defined(ARCH_32BIT)
+        start = 0xb7792000;
+        end   = 0xb7794000;
+#endif
 
         can_read    = true;
         can_write   = false;
@@ -117,8 +141,13 @@ TEST_CASE("Parse maps", "[task][maps]")
 
     SECTION("Pathname with spaces")
     {
+#if defined(ARCH_64BIT)
         start = 0x7f3fcf467000;
         end   = 0x7f3fcf666000;
+#elif defined(ARCH_32BIT)
+        start = 0xb737e000;
+        end   = 0xb7384000;
+#endif
 
         can_read    = false;
         can_write   = false;

--- a/test/mem.cpp
+++ b/test/mem.cpp
@@ -1,15 +1,18 @@
 #include "catch.hpp"
+#include <iostream>
 
 #include "pfs/procfs.hpp"
 
 TEST_CASE("Read mem", "[mem][read]")
 {
     std::vector<uint8_t> secret = {'s', 'e', 'c', 'r', 'e', 't'};
-    off_t secret_offset         = reinterpret_cast<off_t>(&secret[0]);
+
+    // Cast first to size_t to avoid sign expansion of the pointer value
+    size_t secret_offset = reinterpret_cast<size_t>(secret.data());
 
     auto self      = pfs::procfs().get_task();
     auto mem       = self.get_mem();
-    auto extracted = mem.read(secret_offset, secret.size());
+    auto extracted = mem.read(static_cast<loff_t>(secret_offset), secret.size());
 
     REQUIRE(secret == extracted);
 }

--- a/test/meminfo.cpp
+++ b/test/meminfo.cpp
@@ -1,4 +1,7 @@
+#include <sstream>
+
 #include "catch.hpp"
+#include "test_utils.hpp"
 
 #include "pfs/parsers.hpp"
 
@@ -14,36 +17,40 @@ TEST_CASE("Parse corrupted meminfo", "[procfs][meminfo]")
 
 TEST_CASE("Parse meminfo", "[procfs][meminfo]")
 {
-    std::string line;
+    std::stringstream line;
 
     std::string description;
     size_t amount;
 
     SECTION("With kB")
     {
-        line = "VmallocTotal:   34359738367 kB";
-
         description = "VmallocTotal";
+#if defined(ARCH_64BIT)
         amount      = 34359738367;
+#elif defined(ARCH_32BIT)
+        amount      = 512056;
+#endif
+
+        line << description << ":   " << amount << " kB";
     }
 
     SECTION("Zero with kB")
     {
-        line = "VmallocUsed:           0 kB";
-
         description = "VmallocUsed";
         amount      = 0;
+
+        line << description << ":           " << amount << " kB";
     }
 
     SECTION("Zero without kB")
     {
-        line = "HugePages_Total:       0";
-
         description = "HugePages_Total";
         amount      = 0;
+
+        line << description << ":       " << amount;
     }
 
-    auto output = parse_meminfo_line(line);
+    auto output = parse_meminfo_line(line.str());
     REQUIRE(output.first == description);
     REQUIRE(output.second == amount);
 }

--- a/test/modules.cpp
+++ b/test/modules.cpp
@@ -1,4 +1,7 @@
+#include <sstream>
+
 #include "catch.hpp"
+#include "test_utils.hpp"
 
 #include "pfs/parsers.hpp"
 
@@ -15,55 +18,79 @@ TEST_CASE("Parse corrupted modules", "[procfs][modules]")
 TEST_CASE("Parse modules", "[procfs][modules]")
 {
     pfs::module expected;
-    std::string line;
+    std::ostringstream line;
+    line << std::hex;
 
     SECTION("No dependencies")
     {
-        line = "raid1 40960 0 - Live 0xffffffffc03eb000";
+#if defined(ARCH_64BIT)
+        size_t offset = 0xffffffffc03eb000;
+#elif defined(ARCH_32BIT)
+        size_t offset = 0xe089a000;
+#endif
+
+        line << "raid1 40960 0 - Live 0x" << offset;
 
         expected.name           = "raid1";
         expected.size           = 40960;
         expected.instances      = 0;
         expected.dependencies   = {};
         expected.module_state   = pfs::module::state::live;
-        expected.offset         = 0xffffffffc03eb000;
+        expected.offset         = offset;
         expected.is_out_of_tree = false;
         expected.is_unsigned    = false;
     }
 
     SECTION("Unsigned and Out-of-tree")
     {
-        line = "vboxsf 77824 2 - Live 0xffffffffc0759000 (OE)";
+#if defined(ARCH_64BIT)
+        size_t offset = 0xffffffffc0759000;
+#elif defined(ARCH_32BIT)
+        size_t offset = 0xe0ac1000;
+#endif
+
+        line << "vboxsf 77824 2 - Live 0x" << offset << " (OE)";
 
         expected.name           = "vboxsf";
         expected.size           = 77824;
         expected.instances      = 2;
         expected.dependencies   = {};
         expected.module_state   = pfs::module::state::live;
-        expected.offset         = 0xffffffffc0759000;
+        expected.offset         = offset;
         expected.is_out_of_tree = true;
         expected.is_unsigned    = true;
     }
 
     SECTION("Single dependency")
     {
-        line = "libcrc32c 16384 1 raid456, Live 0xffffffffc03b6000";
+#if defined(ARCH_64BIT)
+        size_t offset = 0xffffffffc03b6000;
+#elif defined(ARCH_32BIT)
+        size_t offset = 0xe089f000;
+#endif
+
+        line << "libcrc32c 16384 1 raid456, Live 0x" << offset;
 
         expected.name           = "libcrc32c";
         expected.size           = 16384;
         expected.instances      = 1;
         expected.dependencies   = {"raid456"};
         expected.module_state   = pfs::module::state::live;
-        expected.offset         = 0xffffffffc03b6000;
+        expected.offset         = offset;
         expected.is_out_of_tree = false;
         expected.is_unsigned    = false;
     }
 
     SECTION("Multi dependencies")
     {
-        line =
-            "raid6_pq 114688 4 btrfs,raid456,async_raid6_recov,async_pq, Live "
-            "0xffffffffc03f6000";
+#if defined(ARCH_64BIT)
+        size_t offset = 0xffffffffc03f6000;
+#elif defined(ARCH_32BIT)
+        size_t offset = 0xe081b000;
+#endif
+
+        line << "raid6_pq 114688 4 btrfs,raid456,async_raid6_recov,async_pq, Live 0x"
+             << offset;
 
         expected.name           = "raid6_pq";
         expected.size           = 114688;
@@ -71,13 +98,13 @@ TEST_CASE("Parse modules", "[procfs][modules]")
         expected.dependencies   = {"btrfs", "raid456", "async_raid6_recov",
                                  "async_pq"};
         expected.module_state   = pfs::module::state::live;
-        expected.offset         = 0xffffffffc03f6000;
+        expected.offset         = offset;
         expected.is_out_of_tree = false;
         expected.is_unsigned    = false;
         ;
     }
 
-    auto module = parse_modules_line(line);
+    auto module = parse_modules_line(line.str());
     REQUIRE(module.name == expected.name);
     REQUIRE(module.size == expected.size);
     REQUIRE(module.instances == expected.instances);

--- a/test/net_socket.cpp
+++ b/test/net_socket.cpp
@@ -1,4 +1,7 @@
+#include <sstream>
+
 #include "catch.hpp"
+#include "test_utils.hpp"
 
 #include "pfs/parsers.hpp"
 
@@ -17,13 +20,21 @@ TEST_CASE("Parse corrupted net socket", "[net][net_socket]")
 TEST_CASE("Parse net socket", "[net][net_socket]")
 {
     pfs::net_socket expected;
-    std::string line;
+
+    std::stringstream line;
+    line << std::hex;
 
     SECTION("IPv4")
     {
-        line =
+#if defined(ARCH_64BIT)
+        size_t skbuff = 0xffff9f55b1420800;
+#elif defined(ARCH_32BIT)
+        size_t skbuff = 0xd8658c00;
+#endif
+
+        line <<
             "1: 3500007F:0035 00000000:0000 0A 00000000:00000000 00:00000000 "
-            "00000000   101        0 15979 1 ffff9f55b1420800 100 0 0 10 0";
+            "00000000   101        0 15979 1 " << skbuff << " 100 0 0 10 0";
 
         expected.slot                 = 1;
         expected.local_ip             = pfs::ip(0x3500007F);
@@ -40,15 +51,21 @@ TEST_CASE("Parse net socket", "[net][net_socket]")
         expected.timeouts             = 0;
         expected.inode                = 15979;
         expected.ref_count            = 1;
-        expected.skbuff               = 0xffff9f55b1420800;
+        expected.skbuff               = skbuff;
     }
 
     SECTION("IPv6")
     {
-        line = "5: 00000000000000000000000000000000:006F "
-               "00000000000000000000000000000000:0000 0A 00000000:00000000 "
-               "00:00000000 00000000     0        0 15737 1 ffff9f55bdb91980 "
-               "100 0 0 10 0";
+#if defined(ARCH_64BIT)
+        size_t skbuff = 0xffff9f55bdb91980;
+#elif defined(ARCH_32BIT)
+        size_t skbuff = 0xdb2b0000;
+#endif
+
+        line << "5: 00000000000000000000000000000000:006F "
+                "00000000000000000000000000000000:0000 0A 00000000:00000000 "
+                "00:00000000 00000000     0        0 15737 1 " << skbuff << " "
+                "100 0 0 10 0";
 
         expected.slot     = 5;
         expected.local_ip = pfs::ip(
@@ -67,10 +84,10 @@ TEST_CASE("Parse net socket", "[net][net_socket]")
         expected.timeouts             = 0;
         expected.inode                = 15737;
         expected.ref_count            = 1;
-        expected.skbuff               = 0xffff9f55bdb91980;
+        expected.skbuff               = skbuff;
     }
 
-    auto socket = parse_net_socket_line(line);
+    auto socket = parse_net_socket_line(line.str());
     REQUIRE(socket.slot == expected.slot);
     REQUIRE(socket.local_ip == expected.local_ip);
     REQUIRE(socket.local_port == expected.local_port);

--- a/test/netlink_socket.cpp
+++ b/test/netlink_socket.cpp
@@ -1,4 +1,7 @@
+#include <sstream>
+
 #include "catch.hpp"
+#include "test_utils.hpp"
 
 #include "pfs/parsers.hpp"
 
@@ -8,7 +11,11 @@ TEST_CASE("Parse corrupted netlink socket", "[net][netlink_socket]")
 {
     // Missing last token (drops)
     std::string line =
+#if defined(ARCH_64BIT)
         "ffff9f55ba8e8800 9   4280338379 00000000 0        0        0 2";
+#elif defined(ARCH_32BIT)
+        "dd1a1c00 0   0      00000000 0        0        0 2";
+#endif
 
     REQUIRE_THROWS_AS(parse_netlink_socket_line(line), pfs::parser_error);
 }
@@ -16,14 +23,21 @@ TEST_CASE("Parse corrupted netlink socket", "[net][netlink_socket]")
 TEST_CASE("Parse netlink socket", "[net][netlink_socket]")
 {
     pfs::netlink_socket expected;
-    std::string line;
+    std::ostringstream line;
+    line << std::hex;
 
     SECTION("High port id")
     {
-        line = "ffff9f55ba8e8800 9   4280338379 00000000 0        0        0 2 "
-               "       0        14098";
+#if defined(ARCH_64BIT)
+        size_t skbuff = 0xffff9f55ba8e8800;
+#elif defined(ARCH_32BIT)
+        size_t skbuff = 0xdb313000;
+#endif
 
-        expected.skbuff    = 0xffff9f55ba8e8800;
+        line << skbuff << " 9   4280338379 00000000 0        0        0 2 "
+                "       0        14098";
+
+        expected.skbuff    = skbuff;
         expected.protocol  = 9;
         expected.port_id   = 4280338379;
         expected.groups    = 0;
@@ -35,12 +49,41 @@ TEST_CASE("Parse netlink socket", "[net][netlink_socket]")
         expected.inode     = 14098;
     }
 
+    SECTION("Zero skbuff")
+    {
+#if defined(ARCH_64BIT)
+        std::string skbuff("0000000000000000");
+#elif defined(ARCH_32BIT)
+        std::string skbuff("00000000");
+#endif
+
+        line << skbuff << " 0   464    00000000 0        0        0 2 "
+                "       0        15644";
+
+        expected.skbuff    = 0;
+        expected.protocol  = 0;
+        expected.port_id   = 464;
+        expected.groups    = 0;
+        expected.rmem      = 0;
+        expected.wmem      = 0;
+        expected.dumping   = false;
+        expected.ref_count = 2;
+        expected.drops     = 0;
+        expected.inode     = 15644;
+    }
+
     SECTION("Hex groups")
     {
-        line = "0000000000000000 0   464    000405d1 0        0        0 2 "
-               "       0        15644";
+#if defined(ARCH_64BIT)
+        size_t skbuff = 0xffff9f55ba8e8800;
+#elif defined(ARCH_32BIT)
+        size_t skbuff = 0xdb313000;
+#endif
 
-        expected.skbuff    = 0x0;
+        line << skbuff << " 0   464    000405d1 0        0        0 2 "
+                "       0        15644";
+
+        expected.skbuff    = skbuff;
         expected.protocol  = 0;
         expected.port_id   = 464;
         expected.groups    = 0x405d1;
@@ -54,10 +97,16 @@ TEST_CASE("Parse netlink socket", "[net][netlink_socket]")
 
     SECTION("Without inode")
     {
-        line = "ffff880037c4a800 15  -4167  00000002 0        0        1 2     "
-               "   0";
+#if defined(ARCH_64BIT)
+        size_t skbuff = 0xffff880037c4a800;
+#elif defined(ARCH_32BIT)
+        size_t skbuff = 0xdd238800;
+#endif
 
-        expected.skbuff    = 0xffff880037c4a800;
+        line << skbuff << " 15  -4167  00000002 0        0        1 2     "
+                "   0";
+
+        expected.skbuff    = skbuff;
         expected.protocol  = 15;
         expected.port_id   = -4167;
         expected.groups    = 2;
@@ -71,10 +120,16 @@ TEST_CASE("Parse netlink socket", "[net][netlink_socket]")
 
     SECTION("Without inode and null dumping")
     {
-        line = "ffff880037c10800 0   4195519 00000000 0        0        (null) "
-               "2        0";
+#if defined(ARCH_64BIT)
+        size_t skbuff = 0xffff880037c10800;
+#elif defined(ARCH_32BIT)
+        size_t skbuff = 0xdd239c00;
+#endif
 
-        expected.skbuff    = 0xffff880037c10800;
+        line << skbuff << " 0   4195519 00000000 0        0        (null) "
+                "2        0";
+
+        expected.skbuff    = skbuff;
         expected.protocol  = 0;
         expected.port_id   = 4195519;
         expected.groups    = 0;
@@ -86,7 +141,7 @@ TEST_CASE("Parse netlink socket", "[net][netlink_socket]")
         expected.inode     = pfs::INVALID_INODE;
     }
 
-    auto socket = parse_netlink_socket_line(line);
+    auto socket = parse_netlink_socket_line(line.str());
     REQUIRE(socket.skbuff == expected.skbuff);
     REQUIRE(socket.protocol == expected.protocol);
     REQUIRE(socket.port_id == expected.port_id);

--- a/test/test_utils.hpp
+++ b/test/test_utils.hpp
@@ -3,6 +3,7 @@
 
 #include <stdlib.h>
 #include <unistd.h>
+#include <stdint.h>
 
 #include <linux/kdev_t.h>
 
@@ -10,6 +11,14 @@
 #include <sstream>
 
 #include "pfs/defer.hpp"
+
+#if INTPTR_MAX == INT64_MAX
+    #define ARCH_64BIT
+#elif INTPTR_MAX == INT32_MAX
+    #define ARCH_32BIT
+#else
+    #error Unsupported pointer size or missing size macros
+#endif
 
 template <typename T>
 inline T generate_random()

--- a/test/unix_socket.cpp
+++ b/test/unix_socket.cpp
@@ -1,4 +1,7 @@
+#include <sstream>
+
 #include "catch.hpp"
+#include "test_utils.hpp"
 
 #include "pfs/parsers.hpp"
 
@@ -7,7 +10,11 @@ using namespace pfs::impl::parsers;
 TEST_CASE("Parse corrupted unix socket", "[net][unix_socket]")
 {
     // Missing last token (inode)
+#if defined(ARCH_64BIT)
     std::string line = "ffff8db2f3e09400: 00000002 00000000 00000000 0002 01";
+#elif defined(ARCH_32BIT)
+    std::string line = "db2eab00: 00000002 00000000 00000000 0002 01";
+#endif
 
     REQUIRE_THROWS_AS(parse_unix_socket_line(line), pfs::parser_error);
 }
@@ -15,13 +22,20 @@ TEST_CASE("Parse corrupted unix socket", "[net][unix_socket]")
 TEST_CASE("Parse unix socket", "[net][unix_socket]")
 {
     pfs::unix_socket expected;
-    std::string line;
+    std::ostringstream line;
+    line << std::hex;
 
     SECTION("Anonymous")
     {
-        line = "ffff8db2f3e09400: 00000002 00000000 00000000 0002 01 21401";
+#if defined(ARCH_64BIT)
+        size_t skbuff = 0xffff8db2f3e09400;
+#elif defined(ARCH_32BIT)
+        size_t skbuff = 0xdc035080;
+#endif
 
-        expected.skbuff       = 0xffff8db2f3e09400;
+        line << skbuff << ": 00000002 00000000 00000000 0002 01 21401";
+
+        expected.skbuff       = skbuff;
         expected.ref_count    = 0x00000002;
         expected.protocol     = 0x00000000;
         expected.flags        = 0x00000000;
@@ -32,10 +46,16 @@ TEST_CASE("Parse unix socket", "[net][unix_socket]")
 
     SECTION("File backed")
     {
-        line = "ffff8db2fd23a000: 00000003 00000000 00000000 0001 03 17031 "
-               "/run/systemd/journal/stdout";
+#if defined(ARCH_64BIT)
+        size_t skbuff = 0xffff8db2fd23a000;
+#elif defined(ARCH_32BIT)
+        size_t skbuff = 0xd9cdadc0;
+#endif
 
-        expected.skbuff       = 0xffff8db2fd23a000;
+        line << skbuff << ": 00000003 00000000 00000000 0001 03 17031 "
+                "/run/systemd/journal/stdout";
+
+        expected.skbuff       = skbuff;
         expected.ref_count    = 0x00000003;
         expected.protocol     = 0x00000000;
         expected.flags        = 0x00000000;
@@ -47,10 +67,16 @@ TEST_CASE("Parse unix socket", "[net][unix_socket]")
 
     SECTION("Abstract namespace socket")
     {
-        line = "ffff880037a393c0: 00000002 00000000 00000000 0002 01  "
-               "9050 @/org/kernel/udev/udevd";
+#if defined(ARCH_64BIT)
+        size_t skbuff = 0xffff880037a393c0;
+#elif defined(ARCH_32BIT)
+        size_t skbuff = 0xdef6cb00;
+#endif
 
-        expected.skbuff       = 0xffff880037a393c0;
+        line << skbuff << ": 00000002 00000000 00000000 0002 01  "
+                "9050 @/org/kernel/udev/udevd";
+
+        expected.skbuff       = skbuff;
         expected.ref_count    = 0x00000002;
         expected.protocol     = 0x00000000;
         expected.flags        = 0x00000000;
@@ -60,7 +86,7 @@ TEST_CASE("Parse unix socket", "[net][unix_socket]")
         expected.path         = "@/org/kernel/udev/udevd";
     }
 
-    auto socket = parse_unix_socket_line(line);
+    auto socket = parse_unix_socket_line(line.str());
     REQUIRE(socket.skbuff == expected.skbuff);
     REQUIRE(socket.ref_count == expected.ref_count);
     REQUIRE(socket.flags == expected.flags);

